### PR TITLE
Dynamic str_replace return type

### PR DESCRIPTION
**Pull request by b1rdex**
Str_replace returns array if the subject is an array and returns string otherwise. So my intention is to set return type based on the third argument type. Does this work?

### CODE REVIEW

**Comment by KapitanOczywisty** [comment]
Shouldn't be `type` instead of `elementType`?

File: meta/.phpstorm.meta.php [hunk 1]
```diff
@@ -81,6 +81,7 @@ function elementType($argNum) {
   override(\array_values(0), type(0));
   override(\array_combine(0), type(1));
 
+  override(\str_replace(0), elementType(2));
```
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
  override(\str_replace(0), type(2));
```

**Comment by b1rdex** [comment]
> Shouldn't be `type` instead of `elementType`?

Yep, I think you're right. Thanks, fixed.

**Comment by orklah** [comment]
Cool! Maybe we could add preg_replace too?

**Comment by b1rdex** [comment]
> Cool! Maybe we could add preg_replace too?

I'm waiting for JetBrains team response, still not sure if it works or not.

**Comment by isfedorov** [comment]
@b1rdex Thank you for your contribution! We will merge your change but unfortunately due to https://youtrack.jetbrains.com/issue/WI-36161 scalar return type still will not be inferred using meta.
Regarding preg_replace_callback and preg_filter please file a separate PR.

**Pull request merged**